article_maker.cc

```diff
@@ -179,56 +135,37 @@ std::string ArticleMaker::makeHtmlHeader( QString const & word,
-            "function processIframeMouseOut() { overIframeId = null; top.focus(); }"
-            "function processIframeMouseOver( newId ) { overIframeId = newId; }"
-            "function processIframeClick() { if( overIframeId != null ) { overIframeId = overIframeId.replace( 'gdexpandframe-', '' ); gdMakeArticleActive( overIframeId ) } }"
-            "function init() { window.addEventListener('blur', processIframeClick, false); }"
-            "window.addEventListener('load', init, false);"
-            "function gdExpandOptPart( expanderId, optionalId ) {  var d1=document.getElementById(expanderId); var i = 0; if( d1.alt == '[+]' ) {"
-            "d1.alt = '[-]'; d1.src = 'qrcx://localhost/icons/collapse_opt.png'; for( i = 0; i < 1000; i++ ) { var d2=document.getElementById( optionalId + i ); if( !d2 ) break; d2.style.display='inline'; } }"
-            "else { d1.alt = '[+]'; d1.src = 'qrcx://localhost/icons/expand_opt.png'; for( i = 0; i < 1000; i++ ) { var d2=document.getElementById( optionalId + i ); if( !d2 ) break; d2.style.display='none'; } } };"
-            "function gdExpandArticle( id ) { elem = document.getElementById('gdarticlefrom-'+id); ico = document.getElementById('expandicon-'+id); art=document.getElementById('gdfrom-'+id);"
-            "ev=window.event; t=null;"
-            "if(ev) t=ev.target || ev.srcElement;"
-            "if(elem.style.display=='inline' && t==ico) {"
-            "elem.style.display='none'; ico.className='gdexpandicon';"
-            "art.className = art.className+' gdcollapsedarticle';"
-            "nm=document.getElementById('gddictname-'+id); nm.style.cursor='pointer';"
-            "if(ev) ev.stopPropagation(); ico.title=''; nm.title=\"";
-  result += tr( "Expand article" ).toUtf8().data();
-  result += "\" } else if(elem.style.display=='none') {"
-            "elem.style.display='inline'; ico.className='gdcollapseicon';"
-            "art.className=art.className.replace(' gdcollapsedarticle','');"
-            "nm=document.getElementById('gddictname-'+id); nm.style.cursor='default';"
-            "nm.title=''; ico.title=\"";
-  result += tr( "Collapse article").toUtf8().data();
-  result += "\" } }"
-            "function gdCheckArticlesNumber() {"
-            "elems=document.getElementsByClassName('gddictname');"
-            "if(elems.length == 1) {"
-              "el=elems.item(0); s=el.id.replace('gddictname-','');"
-              "el=document.getElementById('gdfrom-'+s);"
-              "if(el && el.className.search('gdcollapsedarticle')>0) gdExpandArticle(s);"
-            "} }"
+            "function tr(key) {"
+            " var tr_map = {"
+            "\"Expand article\":\"";
+  result += tr("Expand article").toUtf8().data();
+  result += "\",\"Collapse article\":\"";
+  result += tr("Collapse article").toUtf8().data();
+  result += "\"  };"
+            "return tr_map[key] || '';"
+            "}"
             "</script>";
-
```

Comment:
Why did ArticleMaker need so many changes? Is WebKit==>WebEngine really need all of them? If true, WHY?